pyiron_base/project/project_browser.py

```diff
@@ -0,0 +1,337 @@
+                button.disabled = True
+                if self._hide_path:
+                    button.layout.display = 'none'
+            buttons.append(button)
+        button = widgets.Button(icon="fa-home",
+                                tooltip=self._initial_project_path,
+                                layout=widgets.Layout(width='auto'))
+        button.style.button_color = home_color
+        button.path = self._initial_project
+        if self.fix_path:
+            button.disabled = True
+        button.on_click(on_click)
+        buttons.append(button)
+        buttons.reverse()
+        box.children = tuple(buttons)
+
+    def _on_click_file(self, filename):
+        filepath = os.path.join(self.path, filename)
+        self.output.clear_output(True)
+        try:
+            data = self.project[filename]
+        except(KeyError, IOError):
+            data = None
+        with self.output:
+            if data is not None and str(type(data)).split('.')[0] == "<class 'PIL":
+                try:
+                    data_cp = data.copy()
+                    data_cp.thumbnail((800, 800))
+                except:
+                    data_cp = data
+                display(data_cp)
+            elif data is not None:
+                display(data)
+            else:
+                print([filename])
+        if filepath in self._clickedFiles:
+            self._data = None
+            self._clickedFiles.remove(filepath)
+        else:
+            if data is not None:
+                self._data = FileData(data=data, file=filename, metadata={"path": filepath})
+            # self._clickedFiles.append(filepath)
+            self._clickedFiles = [filepath]
+
+    def _update_filebox(self, filebox):
+        # color definitions
+        dir_color = '#9999FF'
+        node_color = '#9999EE'
+        file_chosen_color = '#FFBBBB'
+        file_color = '#DDDDDD'
+
+        if self._node_as_dirs:
+            dirs = self.dirs + self.nodes
+            files = self.files
+        else:
+            files = self.nodes + self.files
+            dirs = self.dirs
+
+        def on_click_group(b):
+            if self._busy_check():
+                return
+            path = os.path.join(self.path, b.description)
+            self._update_project(path)
+            self._busy_check(False)
+
+        def on_click_file(b):
+            if self._busy_check():
+                return
+            self._on_click_file(b.description)
+            self._update_filebox(filebox)
+            self._busy_check(False)
+
+        buttons = []
+        item_layout = widgets.Layout(width='80%',
+                                     height='30px',
+                                     min_height='24px',
+                                     display='flex',
+                                     align_items="center",
+                                     justify_content='flex-start')
+
+        for f in dirs:
```

Comment:
soft recommendation: I think the entire loops could use reduction into well-named methods.